src/offline/cable_checks.F90

```diff
@@ -50,141 +52,561 @@ MODULE cable_checks_module
+     END IF
+
+  END SUBROUTINE check_range_r2_d2
+
+  SUBROUTINE check_range_r1_d2(vname, parameter_r1, parameter_range, ktau, met)
+
+     CHARACTER(LEN=*) :: vname
+     INTEGER, INTENT(IN) :: ktau
+     TYPE(met_type), INTENT(IN) :: met
+
+     REAL, DIMENSION(:,:), INTENT(IN) :: parameter_r1
+     REAL, DIMENSION(2), INTENT(IN) :: parameter_range
+
+     REAL :: max_val, min_val
+     INTEGER, DIMENSION(2) :: index
+
+     index = 0
+     max_val = MAXVAL(parameter_r1)
+     min_val = MINVAL(parameter_r1)
+
+     IF (min_val < parameter_range(1)) THEN
+          index = FINDLOC(parameter_r1, min_val)
+     END IF
+
+     IF (max_val > parameter_range(2)) THEN
+          index = FINDLOC(parameter_r1, max_val)
+     END IF
+
+     IF (index(1) > 0) THEN
+          CALL range_abort(vname, ktau, met, parameter_r1(index(1), index(2)), &
+                parameter_range, index(1), & 
+          patch(index(1))%latitude, patch(index(1))%longitude)
+     END IF
+
+  END SUBROUTINE check_range_r1_d2
+
+  SUBROUTINE check_range_r2_d3(vname, parameter_r2, parameter_range, ktau, met)
+
+     CHARACTER(LEN=*) :: vname
+     INTEGER, INTENT(IN) :: ktau
+     TYPE(met_type), INTENT(IN) :: met
+
+     REAL(r_2), DIMENSION(:,:,:), INTENT(IN) :: parameter_r2
+     REAL, DIMENSION(2), INTENT(IN) :: parameter_range
+
+     INTEGER, DIMENSION(3) :: index
+     REAL(r_2) :: max_val, min_val
+
+     index = 0
+     max_val = MAXVAL(parameter_r2)
+     min_val = MINVAL(parameter_r2)
+
+     IF (min_val < parameter_range(1)) THEN
+          index = FINDLOC(parameter_r2, min_val)
+     END IF
+
+     IF (max_val > parameter_range(2)) THEN
+          index = FINDLOC(parameter_r2, max_val)
+     END IF
+
+     IF (SIZE(index) > 0) THEN
+          CALL range_abort(vname, ktau, met, REAL(parameter_r2(index(0), index(1), index(2))), &
+                parameter_range, index(1), patch(index(1))%latitude, patch(index(1))%longitude)
+     END IF
+     
+  END SUBROUTINE check_range_r2_d3
+
+  SUBROUTINE check_range_r1_d3(vname, parameter_r1, parameter_range, ktau, met)
+
+     CHARACTER(LEN=*) :: vname
+     INTEGER, INTENT(IN) :: ktau
+     TYPE(met_type), INTENT(IN) :: met
+
+     REAL, DIMENSION(:,:,:), INTENT(IN) :: parameter_r1
+     REAL, DIMENSION(2), INTENT(IN) :: parameter_range
+     INTEGER, DIMENSION(3) :: index
+     REAL :: max_val, min_val
+
+     index = 0
+     max_val = MAXVAL(parameter_r1)
+     min_val = MINVAL(parameter_r1)
+
+     IF (min_val < parameter_range(1)) THEN
+          index = FINDLOC(parameter_r1, min_val)
+     END IF
+
+     IF (max_val > parameter_range(2)) THEN
+          index = FINDLOC(parameter_r1, max_val)
+     END IF
+
+     IF (SIZE(index) > 0) THEN
+          CALL range_abort(vname, ktau, met, parameter_r1(index(1), index(2), index(3)), parameter_range, & 
+               index(1), patch(index(1))%latitude, patch(index(1))%longitude)
+     END IF
+
+  END SUBROUTINE check_range_r1_d3
+
+  ! TODO add for integer
```

Comment:
Why add for integer? Is that currently needed by the code? If not, I don't see the point of adding it now. It can be added if it is ever needed.